sys/include/net/gnrc/lwmac/lwmac_init.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *               2016 INRIA
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_lwmac
+ * @{
+ *
+ * @file
+ * @brief       Initialization of LWMAC
+ * @internal
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ */
+
+#ifndef GNRC_LWMAC_INIT_H
+#define GNRC_LWMAC_INIT_H
+
+#include "net/gnrc/netdev.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Initialize an instance of the LWMAC layer
+ *
+ * The initialization starts a new thread that connects to the given netdev
+ * device and starts a link layer event loop.
+ *
+ * @param[in] stack         stack for the control thread
+ * @param[in] stacksize     size of *stack*
+ * @param[in] priority      priority for the thread housing the LWMAC instance
+ * @param[in] name          name of the thread housing the LWMAC instance
+ * @param[in] dev           netdev device, needs to be already initialized
+ *
+ * @return                  PID of LWMAC thread on success
+ * @return                  -EINVAL if creation of thread fails
+ * @return                  -ENODEV if *dev* is invalid
+ */
+kernel_pid_t gnrc_lwmac_init(char *stack, int stacksize, char priority,
```

Comment:
Em, this is to avoid `gnrc_netdev_t` redifinition. Originally, `lwmac.h` and `gnrc/netdev.h` call each other, so If I put this in `lwmac.h`, I need to forward definition of `gnrc_netdev_t` for this function here (`lwmac.h` ), which will lead to `gnrc_netdev_t` redifinition errors reported by CI.